src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -733,6 +733,12 @@ ExecuteSingleModifyTask(QueryDesc *queryDesc, Task *task,
 		bool failOnError = false;
 		int64 currentAffectedTupleCount = 0;
 
+		if (connection->remoteTransaction.transactionFailed)
```

Comment:
Could elaborate a bit more here, e.g.: We proceeded after a BEGIN failure on the current connections, because we still have other, healthy connections. We can skip this connection, which will be marked as inactive if the write succeeds.